biojava-genome/src/test/java/org/biojava/nbio/genome/io/fastq/FastqBuilderTest.java

```diff
@@ -36,6 +36,41 @@ public void testConstructor()
 		Assert.assertNotNull(fastqBuilder);
 	}
 
+	@Test
+	public void testConstructorFastq()
+	{
+		FastqBuilder fastqBuilder = new FastqBuilder()
+			.withDescription("description")
+			.withSequence("sequence")
+			.withQuality("quality_")
+			.withVariant(FastqVariant.FASTQ_SOLEXA);
+
+		Fastq fastq = fastqBuilder.build();
+
+		FastqBuilder fastqBuilder2 = new FastqBuilder(fastq);
+		Assert.assertNotNull(fastqBuilder2);
+
+		Fastq fastq2 = fastqBuilder2.build();
+		Assert.assertEquals("description", fastq2.getDescription());
+		Assert.assertEquals("sequence", fastq2.getSequence());
+		Assert.assertEquals("quality_", fastq2.getQuality());
+		Assert.assertEquals(FastqVariant.FASTQ_SOLEXA, fastq2.getVariant());
+	}
+
+	@Test
+	public void testConstructorNullFastq()
+	{
+		try
+		{
+			new FastqBuilder(null);
+			Assert.fail("builder(null) expected IllegalArgumentException");
+		}
+		catch (IllegalArgumentException e)
+		{
+			// expected
```

Comment:
I think it'd be cleaner to use the `Assert.assertThrows()` call available in junit 4.13+. See https://stackoverflow.com/questions/156503/how-do-you-assert-that-a-certain-exception-is-thrown-in-junit-4-tests/46514550#46514550

Reply:
I agree, and perhaps it'd be even more preferable to use `@Test(expected=IllegalArgumentException.class)` but this stuff is old and I followed the pattern used elsewhere in the same file

Reply:
> more preferable to use @Test(expected=IllegalArgumentException.class)

That seems to be so before 4.13. I just read in the SO thread that from 4.13+, it is preferrable to do `Assert.assertThrows()`. Anyway since these tests are new, it's a good idea to follow the latest best practices.